torch/nn/functional.py

```diff
@@ -2817,14 +2817,15 @@ def gaussian_nll_loss(
         raise ValueError(reduction + " is not a valid value for reduction")
 
     # Clamp for stability
-    var = var.clone()
```

Comment:
not sure what was the purpose of cloning here since the variable is shadowed later anyway

Reply:
Now I know why. It's either the original behavior (with `no_grad` and inplace clamp) or just this (without any context): `var = var.clamp(min=eps)`. I couldn't find any other code that wouldn't break the following tests:

```
python -m pytest test/test_modules.py -k GaussianNLLLoss -vvv
python -m pytest test/test_ops_gradients.py -k gaussian_nll_loss -vvv
```

[This](https://github.com/pytorch/pytorch/issues/53964#issuecomment-798067924) claims that doing it without `no_grad` will "cause divergence," but the tests pass locally.

Reply:
any thoughts @albanD ? Clamping without `no_grad` LGTM, but I know we've historically done it with the no_grad...

Reply:
clamping with `no_grad` will zero out a bunch of gradients. We definitely don't want that!

Reply:
What about figuring out what's the value we want this function to take at `var = 0` and return that? Although we would need to somehow deal with the gradients as well...

Reply:
Note: @lezcano told me offline there are plans (IIUC) to have a "framework" that would help with numerical issues like this one, so postponing for now.